.github/workflows/deploy-prod.yaml

```diff
@@ -1,19 +1,20 @@
-name: Deploy to Production Environment
+name: Deploy to Production
 
 on:
-  workflow_dispatch:
+  # deploy when version tags are published
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
```

Comment:
only run this when a new version tag is uploaded.

in theory this should be the only way to deploy to prod in the future, but in practice we'll probably still need to be able to run manual deploys, at least at the start. for that, use the manual deploy action.